packages/builders/scripts/vats/terminate-governor-instance.js

```diff
@@ -0,0 +1,131 @@
+
+const repr = val =>
+  typeof val === 'string' || (typeof val === 'object' && val !== null)
+    ? JSON.stringify(val)
+    : String(val);
+const defaultMakeError = (strings, ...subs) =>
+  Error(
+    strings.map((s, i) => `${i === 0 ? '' : repr(subs[i - 1])}${s}`).join(''),
+  );
+const makeUsageError = (strings, ...subs) => {
+  const err = defaultMakeError(strings, ...subs);
+  console.error(err.message);
+  console.error(USAGE);
+  return err;
+};
+
+const rtarget = /^(?<boardID>board[0-9]+):(?<instanceKitLabel>.+)$/;
+/**
+ * @param {string[]} args
+ * @param {(strings: TemplateStringsArray | string[], ...subs: unknown[]) => Error} [makeError]
+ * @returns {Array<{boardID: string, instanceKitLabel: string}>}
+ */
+const parseTargets = (args, makeError = defaultMakeError) => {
+  /** @type {Array<{boardID: string, instanceKitLabel: string}>} */
+  const targets = [];
+  const badTargets = [];
+  for (const arg of args) {
+    const m = typeof arg === 'string' && arg.match(rtarget);
+    if (!m) {
+      badTargets.push(arg);
+    } else {
+      // @ts-expect-error cast
+      targets.push(m.groups);
+    }
+  }
+  if (badTargets.length !== 0) {
+    throw makeError`malformed target(s): ${badTargets}`;
+  }
+  return targets;
+};
+
+/**
+ * @param {BootstrapPowers} powers
+ * @param {{ options: { targetSpecifiers: string[] } }} config
+ */
+export const terminateGovernors = async (
+  { consume: { board, governedContractKits } },
+  { options: { targetSpecifiers } },
+) => {
+  const { Fail, quote: q } = assert;
+  const targets = parseTargets(targetSpecifiers, Fail);
+  const doneP = Promise.allSettled(
+    targets.map(async ({ boardID, instanceKitLabel }) => {
+      const logLabel = [boardID, instanceKitLabel];
+      const contractInstanceHandle = await E(board).getValue(boardID);
+      const instanceKit = await E(governedContractKits).get(
+        // @ts-expect-error TS2345 Property '[tag]' is missing
+        contractInstanceHandle,
+      );
+      console.log(
+        `${q(logLabel)} alleged governor contract instance kit`,
+        instanceKit,
+      );
+      const { label, governorAdminFacet, adminFacet } = instanceKit;
+      label === instanceKitLabel ||
+        Fail`${q(logLabel)} unexpected instanceKit label, got ${label} but wanted ${q(instanceKitLabel)}`;
+      (adminFacet && adminFacet !== governorAdminFacet) ||
+        Fail`${q(logLabel)} instanceKit adminFacet should have been present and different from governorAdminFacet but was ${adminFacet}`;
+      const reason = harden(Error(`core-eval terminating ${label} governor`));
+      await E(governorAdminFacet).terminateContract(reason);
+      console.log(`${q(logLabel)} terminated governor`);
+    }),
+  );
+  const results = await doneP;
+  const problems = targets.flatMap(({ boardID, instanceKitLabel }, i) => {
+    if (results[i].status === 'fulfilled') return [];
+    return [[boardID, instanceKitLabel, results[i].reason]];
+  });
```

Comment:
That's an interesting filter + map pattern.

Reply:
Oh yeah, I've got a TODO somewhere to add a helper to `@agoric/internal`. Aside from the slower performance, prettier just makes `.filter(…).map(…)` take up too much real estate.

Reply:
Helper? I wonder why. flatMap is a recognized workhorse idiom, no?